### ISSUE TITLE
Issues compiling fpLLL on a Cray Shasta system

### ISSUE THREAD

**Issue by ekera** [fplll/fplll#483]
I'm trying to compile the current version of fpLLL on a Cray Shasta system.

The login node is running "SLES 15 SP1 SuSE Linux System".

I'm on the <code>master</code> branch at commit e4ea83d15a777363acbbe50a1d0e1814ca8bee69.

I had some initial issues with dependencies, but I manually resolved these by installing autoconf-2.71, automake-1.16.5 and libtool-2.4.6 and adding the relevant path variables etc. I then do the following to compile fpLLL:

```tcsh
uan01 Private/depend> git clone https://github.com/fplll/fplll.git
uan01 Private/depend> cd fplll
uan01 depend/fplll> ./autogen.sh 
:
The fplll build system is now prepared.  To build here, run:
  ./configure
  make
uan01 depend/fplll> ./configure --prefix=/cfs/klemming/home/e/ekera/Private/depend/build-products/fplll --exec-prefix=EPREFIX=/cfs/klemming/home/e/ekera/Private/depend/build-products/fplll
:
checking whether make sets $(MAKE)... (cached) yes
checking whether g++ supports C++11 features with -std=c++11... yes
checking for __gmp_get_memory_functions in -lgmp... yes
checking for mpfr_fms in -lmpfr... yes
./configure: line 18562: syntax error near unexpected token `LIBQD,'
./configure: line 18562: `  PKG_CHECK_MODULES(LIBQD, qd, have_libqd="yes",'
uan01 depend/fplll> 
```

It seems as if <code>autogen.sh</code> is generating a <code>./configure</code> script that contains syntax errors?

I also tried to add e.g. <code>--with-qd=no</code>, <code>--without-qd</code> etc. to the command line as options to attempt to circumvent the error, but this seemingly had no effect. The same error message is produced

**Comment by ekera** [comment]
This may also be related to issue #410.

**Comment by ekera** [comment]
I've also tried

```tcsh
uan01 depend/fplll> git checkout 5.4.1 -b 5.4.1
```

leaving me on commit 80ceff47ade5463e1c4f222552daa4a82824cbbc but the same error arises when configuring.

**Comment by ekera** [comment]
No, I already tried this, and <code>pkg-config</code> is available, at least on the command line:
```tcsh
uan01 depend/fplll> pkg-config
Must specify package names on the command line
```

I did try to checkout release after release and compiling from scratch. It works if I check out 5.3.2 before running <code>autogen.sh</code> and then <code>./configure</code>. This may be done by executing:
```tcsh
uan01 depend/fplll> git checkout 5.4.0 -b 5.4.0
```

From 5.3.3 onwards, the error arises with the <code>configure</code> script containing syntax errors.

**Comment by malb** [comment]
In 5.3.3 we did a bunch of `pkg-config` stuff so that would point in the right direction.

**Comment by malb** [comment]
Is there like a docker image or something with SLES 15 SP1 SuSE Linux System and some instructions of how to get to the failure? I could try to reproduce locally and debug.

**Comment by ekera** [comment]
Since this is a Cray Shasta system ([Dardel](https://www.pdc.kth.se/hpc-services/computing-systems/about-dardel-1.1053338) at [KTH](https://www.kth.se/en)), I think that the software is furnished by HPE Cray. I am not sure if it can easily be downloaded. I suspect it may be a modified version with some proprietary parts to work in the HPC environment. I can ask HPC support and get back to you.

I can also send you the broken configure script by e-mail, if that would be somehow useful in tracking down the bug.

**Comment by ekera** [comment]
For anyone following this thread, we've agreed by e-mail to try to sort this out when time allows, in February or March. This with access to Dardel where the error can be reproduced. Until the problem is fixed, use release v5.3.2. It compiles properly.
